### PR TITLE
[Merged by Bors] - feat: port MeasureTheory.Measure.MeasureSpace

### PR DESCRIPTION
**Pull request by MonadMaverick**
This PR also renames instances in `MeasureTheory.MeasurableSpace`.

---

- [x] depends on: #3351
- [x] depends on: #3349
- [x] depends on: #3800
- [x] depends on: #3816
- [x] depends on: #3325
- [x] depends on: #3674 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#3351
* leanprover-community/mathlib4#3349
* leanprover-community/mathlib4#3325
* leanprover-community/mathlib4#3170
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#3351~~
* ~~leanprover-community/mathlib4#3349~~
* ~~leanprover-community/mathlib4#3800~~
* ~~leanprover-community/mathlib4#3816~~
* ~~leanprover-community/mathlib4#3325~~
* ~~leanprover-community/mathlib4#3674~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/MeasureTheory/Measure/MeasureSpace.lean [hunk 1]
```diff
@@ -0,0 +1,4677 @@
...
+
+@[simp]
+theorem toMeasure_toOuterMeasure (m : OuterMeasure α) (h : ms ≤ m.caratheodory) :
+    (m.toMeasure h).toOuterMeasure = m.trim :=
+  rfl
+#align measure_theory.to_measure_to_outer_measure MeasureTheory.toMeasure_toOuterMeasure
+
+-- Porting note: A coercion is directly elaborated in Lean4, so the LHS is simplified by
+-- `toMeasure_toOuterMeasure` even if this theorem has high priority.
+@[simp, nolint simpNF]
```
Comment (ChrisHughes24) [inline Mathlib/MeasureTheory/Measure/MeasureSpace.lean:667]: For all of these `nolint simpNF` lemmas, could you add a new version of the lemma with the `simp` attribute, the same RHS as this lemma, and the LHS in simp-normal form

**Review by ChrisHughes24 — APPROVED** [review]
One comment otherwise LGTM

File: Mathlib/MeasureTheory/Measure/MeasureSpace.lean (hunk 1 above)
Comment (Komyyy) [reply Mathlib/MeasureTheory/Measure/MeasureSpace.lean]: The theorem correspending `MeasureTheory.Measure.sum_coe_finset` couldn't be simplified.

**Comment by ChrisHughes24** [comment]
Thanks

bors merge

**Pull request closed without merging**